_pytest/pytester.py

```diff
@@ -481,6 +481,7 @@ def _makefile(self, ext, args, kwargs):
         ret = None
         for name, value in items:
             p = self.tmpdir.join(name).new(ext=ext)
+            p.dirpath().ensure_dir()
```

Comment:
No, just that I needed this to make the test easier. Now one can use sub-directories in the keys to `testdir.makepyfile` (actually any "make*" functions):

```python
testdir.makepyfile(**{
    'foo/bar/foobar.py': 'print("hello world")',
})
```